src/nbl/asset/utils/IShaderCompiler.cpp

```diff
@@ -247,24 +259,36 @@ std::vector<uint8_t> IShaderCompiler::CCache::serialize()
     return retVal;
 }
 
-core::smart_refctd_ptr<IShaderCompiler::CCache> IShaderCompiler::CCache::deserialize(core::smart_refctd_ptr<ICPUBuffer> serializedCache)
+core::smart_refctd_ptr<IShaderCompiler::CCache> IShaderCompiler::CCache::deserialize(std::span<uint8_t> serializedCache)
 {
     auto retVal = core::make_smart_refctd_ptr<CCache>();
-    retVal->storageBuffer = std::move(serializedCache);
-    // First get the size of the json in the buffer
-    uint64_t* bufferStart = (uint64_t*)retVal->storageBuffer->getPointer();
-    retVal->containerJsonSize = bufferStart[0];
-    uint8_t* containerJsonStart = (uint8_t*)(bufferStart + 1);
-    std::string containerJsonString(containerJsonStart, containerJsonStart + retVal->containerJsonSize);
+
+    // First get the size of the json in the buffer, stored in the first 8 bytes
+    uint64_t* cacheStart = reinterpret_cast<uint64_t*>(serializedCache.data());
+    uint64_t containerJsonSize = cacheStart[0];
+    // Next up get the json that stores the container data
+    std::string containerJsonString(serializedCache.begin() + CONTAINER_JSON_SIZE_BYTES, serializedCache.begin() + CONTAINER_JSON_SIZE_BYTES + containerJsonSize);
```

Comment:
btw learn about `std::string_view` it prevents copies, just casts any memory range to act as a string